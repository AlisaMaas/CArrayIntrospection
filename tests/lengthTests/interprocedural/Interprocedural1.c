/**
* Basic interprocedural test, nothing fancy.
**/
void print (char c){
}

int bar(char *x, int y) {
    for (int i = 0; i < y; i++) {
       print(x[i]); 
    }
    return 1;
}

int foo(char *x, int y) {
    return bar(x, y);
}