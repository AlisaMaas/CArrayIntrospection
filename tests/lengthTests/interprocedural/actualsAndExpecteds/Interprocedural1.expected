print with argument 0 should be annotated No length value.
bar with argument 0 should be annotated Parameter length of 1.
bar with argument 1 should be annotated No length value.
foo with argument 0 should be annotated Parameter length of 1.
foo with argument 1 should be annotated No length value.
